### PR TITLE
[Proposal] Organize Storybook folder structure

### PR DESCRIPTION
**Pull request by langermank**
This PR accompanies this [Discussion](https://github.com/github/primer/discussions/1305) (hubbers only) about Storybook/Lookbook structure. This is just a start, but with our [prospects of single page component docs](https://github.com/github/primer/issues/1285) Storybook becomes even more critical for documentation. If Storybook is organized closer to our more "official" documentation, it may be easier to navigate and parse. 

### Screenshots

See the preview deploy 😄 

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

### CODE REVIEW

**Comment by broccolinisoup** [comment]
I love how navigating is much more intuitive now @langermank ! 😍 And thanks for doing it, there were quite a lot of files to edit! 

The only thing is that the [lockfileVersion](https://docs.npmjs.com/cli/v8/configuring-npm/package-lock-json#lockfileversion) test is failing. According to the tests, apparently it should be 2 but it changed to 1. Any chance you might be using npm v6?

**Review by broccolinisoup — COMMENTED** [review]
I love how navigating is much more intuitive now @langermank ! 😍 And thanks for doing it, there were quite a lot of files to edit!

The only thing is that the [lockfileVersion](https://docs.npmjs.com/cli/v8/configuring-npm/package-lock-json#lockfileversion) test is failing. According to the tests, apparently it should be 2 but it changed to 1. Any chance you might be using npm v6?

**Review by joshblack — COMMENTED** [review]
Love this! The order makes a lot of sense to me for the different groups and tying things under `Components` as the group title feels clear.

**Comment by langermank** [comment]
@primer/react-reviewers crew: should I add a changeset for this? Not sure since its just docs/storybook related 😄 lmk!

**Review by colebemis — APPROVED** [review]
🚀

File: .storybook/preview.js [hunk 1]
```diff
@@ -16,5 +16,11 @@ export const parameters = {
   html: {
     root: '#html-addon-root',
     removeEmptyComments: true
+  },
+  options: {
+    storySort: {
+      order: ['Components', 'Behaviors', 'Hooks', 'Private components', 'Deprecated components', '*'],
```
Comment (siddharthkp) [inline .storybook/preview.js:22]: nice!

File: package-lock.json [hunk 2]
```diff
@@ -47456,6 +47449,17 @@
         "@storybook/instrumenter": "^6.4.0",
         "@testing-library/jest-dom": "^5.16.2",
         "jest-mock": "^27.3.0"
+      },
+      "dependencies": {
```
Comment (siddharthkp) [inline package-lock.json:47453]: nit, probably nothing: that's a bit strange that this made it to the diff even though there isn't a change in package.json 🤔 

which node + npm version are you on?

File: src/PageLayout/PageLayout.stories.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import {Placeholder} from '../Placeholder'
 import {PageLayout} from './PageLayout'
 
 const meta: Meta = {
-  title: 'Layout/PageLayout',
+  title: 'Components/PageLayout',
```
Comment (siddharthkp) [inline src/PageLayout/PageLayout.stories.tsx:8]: nit, not blocking: do you think we need another layer of nesting? Components/Layouts/PageLayout?

**Review by siddharthkp — APPROVED** [review]
Left a few comments, but nothing blocking. Ship it :shipit:

**Pull request merged**